internal/driver/driver.go

```diff
@@ -28,10 +28,28 @@ type Driver struct {
 	addressMap          map[string]chan bool
 	workingAddressCount map[string]int
 	stopped             bool
+	clientMap           map[string]DeviceClient
 }
 
 var concurrentCommandLimit = 100
 
+func (d *Driver) createDeviceClient(info *ConnectionInfo) (DeviceClient, error) {
+	d.mutex.Lock()
```

Comment:
the mutex for address lock and clientMap should not be the same, because it's may to cause deadlock and hard to debug
use two separated mutex in the driver: `addressMutex` and `clientMutex`

Reply:
Okay, thanks for the guidance.